sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReaderSuite.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.orc
+
+import org.apache.orc.TypeDescription
+
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.vectorized.{OnHeapColumnVector, WritableColumnVector}
+import org.apache.spark.sql.test.{SharedSQLContext, SQLTestUtils}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.unsafe.types.UTF8String.fromString
+
+class OrcColumnarBatchReaderSuite extends QueryTest with SQLTestUtils with SharedSQLContext {
+
+  private val orcFileSchema = TypeDescription.fromString(s"struct<col1:int,col2:int>")
+  private val requiredSchema = StructType.fromDDL("col1 int, col3 int")
+  private val partitionSchema = StructType.fromDDL("partCol1 string, partCol2 string")
+  private val partitionValues = InternalRow(fromString("partValue1"), fromString("partValue2"))
+  private val resultSchema = StructType(requiredSchema.fields ++ partitionSchema.fields)
+
+  private val isConstant = classOf[WritableColumnVector].getDeclaredField("isConstant")
+  isConstant.setAccessible(true)
```

Comment:
@cloud-fan and @gengliangwang 
To avoid changing the visibility of `WritableColumnVector.isConstant`, I use reflection here. I guess this is okay for testing. For `OrcColumnarBatchReader`, I changed the visibility two fields with `VisibleForTesting` annotation.